app/src/main/java/org/mozilla/fenix/components/history/PagedHistoryProvider.kt

```diff
@@ -144,6 +160,12 @@ class DefaultPagedHistoryProvider(
                         excludeTypes = excludedVisitTypes
                     )
                     .map { transformVisitInfoToHistoryItem(it) }
+                    .distinctBy { Pair(it.historyTimeGroup, it.url) }
```

Comment:
Won't it filter out the elements with the same `historyTimeGroup` and `url` when `historyImprovementFeatures == false` ?

Reply:
Good catch.  I'll fix.  Thanks